src/java.management/share/classes/javax/management/loading/MLet.java

```diff
@@ -1144,16 +1144,7 @@ private synchronized String loadLibraryAsResource(String libname) {
                                                       libname + ".", null)
                                       .toFile();
                      file.deleteOnExit();
-                     FileOutputStream fileOutput = new FileOutputStream(file);
-                     try {
-                         byte[] buf = new byte[4096];
-                         int n;
-                         while ((n = is.read(buf)) >= 0) {
-                            fileOutput.write(buf, 0, n);
-                         }
-                     } finally {
-                         fileOutput.close();
-                     }
+                     Files.copy(is, file.toPath());
```

Comment:
One thing to check here is the existing behavior when the file already exists (as Files.copy will fail if the file exists, need to specify REPLACE_EXISTING to get the semantics of the existing code).

The code that follows checks if the file exists, this will always be true when Files.copy succeeds.

Reply:
fixed